File: tests/unit/native_java/src/haxe/test/StaticAndInstanceClash.java
package haxe.test;

public class StaticAndInstanceClash extends Base.InnerClass
{
	public static String someString;

	public static class StaticAndInstanceClashSame extends StaticAndInstanceClash
	{
		public int someInt;
		//thankfully, Java forbids this
		//public static int someInt;
	}

	public static class StaticAndInstanceAndMethodClash extends StaticAndInstanceClashSame
	{
		//thankfully, Java forbids this
		//public static float someFloat;

		//Haxe will ignore this declaration, since @:overload is only meant for methods, not for methods + member variables
		public float someFloat;

		public float someFloat()
		{
			return someFloat;
		}

		public float someFloat(float val)
		{
			return someFloat = val;
		}

		public int someInt()
		{
			return someInt;
		}
	}
}

File: tests/unit/native_cs/src/haxe/test/StaticAndInstanceClash.cs
namespace haxe.test
{

public class StaticAndInstanceClash : Base.InnerClass
{
	~StaticAndInstanceClash() {}
	public static string someString;

	public class StaticAndInstanceClashSame : StaticAndInstanceClash
	{
		public int someInt;
	}

	public class StaticAndInstanceAndMethodClash : StaticAndInstanceClashSame
	{

		private float sfloat = 5;
		private int sint = 5;
		public float someFloat()
		{
			return sfloat;
		}

		public float someFloat(float val)
		{
			return sfloat = val;
		}

		public int someInt()
		{
			return sint;
		}
	}
}

}
